x-pack/plugins/uptime/server/lib/requests/search/refine_potential_matches.ts

```diff
@@ -116,13 +115,39 @@ export const mostRecentCheckGroups = async (
             location: {
               terms: { field: 'observer.geo.name', missing: 'N/A', size: 100 },
               aggs: {
-                top: {
-                  top_hits: {
-                    sort: [{ '@timestamp': 'desc' }],
-                    _source: {
-                      includes: ['monitor.check_group', '@timestamp', 'summary.up', 'summary.down'],
+                summaries: {
+                  // only match summary docs because we only want the latest *complete* check group.
+                  filter: { exists: { field: 'summary' } },
+                  aggs: {
+                    latest: {
+                      top_hits: {
+                        sort: [{ '@timestamp': 'desc' }],
+                        _source: {
+                          includes: [
+                            'monitor.check_group',
+                            '@timestamp',
+                            'summary.up',
+                            'summary.down',
+                          ],
+                        },
+                        size: 1,
+                      },
+                    },
+                  },
+                },
+                // We want to find the latest check group, even if it's not part of a summary
+                latest_matching: {
```

Comment:
so this is why summary filter is moved inside aggs, i believe we can also add filters to terms aggs for obserer.geo.name.

Reply:
I think it'll still be pretty efficient even with this change. I'd say unless we see specific issues in a benchmark it let's go for clarity here.